### PR TITLE
* Added powerstrip support * Fixed issues related to being able to set the interface namespace from within the    container

### PR DESCRIPTION
**Pull request by mrjana**

### CODE REVIEW

File: scripts/socketplane.sh [hunk 1]
```diff
@@ -316,6 +320,12 @@ start_socketplane() {
 
     mkdir -p /var/run/socketplane
     echo $cid > /var/run/socketplane/cid
+
+    if [ "$ps" = "yes" ]; then
+	pscid=$(docker run -d --name powerstrip -v /var/run/docker.sock:/var/run/docker.sock \
+	    -v /etc/socketplane/adapters.yml:/etc/powerstrip/adapters.yml --net=host  \
```
Comment (mavenugo) [inline scripts/socketplane.sh]: This seems specific to the vagrant setup. Shouldnt this be applicable to non-vagrant setup as well ?

File: daemon/api.go [hunk 2]
```diff
@@ -90,12 +128,107 @@ func createRouter(d *Daemon) *mux.Router {
...
+		data, _ = json.Marshal(psAdapterPreHook(d, reqParams))
+	case "post-hook":
+		data, _ = json.Marshal(psAdapterPostHook(d, reqParams))
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.Write(data)
+	return nil
+}
+
```
Comment (mavenugo) [inline daemon/api.go]: It would be awesome to move all powerstrip specific code to powerstrip.go.

File: scripts/socketplane.sh (hunk 1 above)
Comment (mrjana) [inline scripts/socketplane.sh]: Hmm.. I see your point. But if we want to make it seamless we need make an assumption on where certain pieces go. Where is the socketplane script placed in a non-vagrant setup?
We can just say that place the repo .yml file in /etc/socketplane/adapters.yml

Comment (mavenugo) [inline scripts/socketplane.sh]: Ya. We could state that .yml file should be in /etc/socketplane/adapters.yml.
But, there are 2 cases we can handle it automatically.
* Developer machine. We could potentially copy the adapters.yml in the local directory and copy it to /etc/socketplane
* if the users uses the install.sh (as specified in the README.md), then we can fetch the adapters.yml from the github repo. (We do the same today for socketplane.sh and place it in /usr/bin)

**Comment by mavenugo** [comment]
@mrjana Thanks for addressing the comments. am trying to get it working in a dev environment without using vagrant or install script. No luck yet. I will try Vagrant and install script shortly and provide feedback.

In the meantime, is it possible for you to squash the first 3 commits (8f14953d9431d1f0ab2a314d2873b53888f9a1f7, f8fde7b51a71ab584a09cab0d351cfe2e921df3a and 6562b6b98c3e844d4c9bc8e81f9281cdeed35c35) into a single commit so that the commits are individually consumable ?

**Pull request closed without merging**